src/spec-node/dockerCompose.ts

```diff
@@ -244,6 +244,25 @@ ${cacheFromOverrideContent}
 		args.push('-f', composeOverrideFile);
 	}
 
+	const hasGpuRequirement = config.hostRequirements?.gpu;
+	const addGpuCapability = hasGpuRequirement && await checkDockerSupportForGPU(params);
+	if (addGpuCapability) {
+		const composeOverrideContent = `services:
+  ${config.service}:
```

Comment:
`generateFeaturesComposeOverrideContent()` would be the better spot for adding this. The function name is increasingly misleading, but that's the override file for everything we need to create the container. (The code here takes care of the building.)

Reply:
I've moved the logic into that function - it reduces the complexity significantly - thanks :)